### ISSUE TITLE
va-accordion is not 508 compliant

### ISSUE THREAD

**Issue by imelda-guzman** [department-of-veterans-affairs/vets-design-system-documentation#1352]
# Bug Report

- [X] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

Our application recently went through a 508 audit and came back with several defects. One of the defects was reported against the va-accordion header/title.

![va-accordion-non-508-compliant](https://user-images.githubusercontent.com/748095/203653947-408b25a0-c1a4-44fa-9208-e831ec01c09c.jpeg)

**Defect Criterion - 1.3.1 Info and Relationships**

- Throughout the product, the correct heading structure cannot be understood by assistive technology. For Example, on the VA Self Scheduling screen (Home> VA Self Scheduling) assistive technology (JAWS) reads ‘"Outpatient Mental Health (502) heading level 2"’, however ‘Outpatient Mental Health’ and ‘(502)’ are not visually equivalent.

## What I expected to happen
Per 508 feedback:
Not to hear the accordion header/title be read back as "...heading level 2".  (There is no h2 on the page for "Outpatient Mental Health (502)")

## Reproducing
- Component Library version: 12.1.0
- Formation version: 7.0.2
- Web Components version: 4.15.0
- Device: HP Elitebook
- Browser: Edge, Chrome

Steps to reproduce:

1. Add the va-accordion to a page
2. Use a screen reader (JAWS) and tab to accordion item.
3. Listen for the name of the accordion header and note that screen reader reads out "...heading level 2"

## Urgency
How urgent is this request? Please select the appropriate option below and/or provide details
- [ ] This bug is blocking work currently in progress
- [X] This bug is affecting work currently in progress, but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

### Details
Please let me know if there is a way to remedy this issue for the va-accordion or if you would like contact information for our 508 representative.

**Comment by jamigibbs** [comment]
Hi @imelda-guzman. Can you provide a link to where this accordion is implemented? The header is passed into the component as a slot so there could be a number of ways that has been done. [For example](https://design.va.gov/storybook/?path=/docs/components-va-accordion--default):

```html
  <va-accordion-item id="first">
    <h6 slot="headline">
      First Amendment Headline
    </h6>
    Congress shall make no law respecting an establishment of religion, or prohibiting the free exercise thereof; or abridging the freedom of speech, or of the press; or the right of the people peaceably to assemble, and to petition the Government for a redress of grievances.
  </va-accordion-item>
```

**Comment by SarahKay8** [comment]
@imelda-guzman Thanks for bringing this to our attention. After receiving a link, I'd like to run a few tests on it.

**Comment by imelda-guzman** [comment]
@jamigibbs our application is not in production yet only in our dev and staging environment. 
We are currently using the va-accordion in our code like this:
```
<va-accordion-item header="Mandatory Fields" subheader="(502)">
   content...
</va-accordion-item>
```
It appears the "header" attribute defines what gets printed in the slot.  According to documentation in [Storybook ](https://design.va.gov/storybook/?path=/docs/components-va-accordion--default) I can change the level of the header but given the feedback we received from the 508 team I don't think we want these accordion items to be read back as headers.

**Comment by imelda-guzman** [comment]
Hi @caw310 I'm following up to see if your team was able to test and find a solution for screen readers.

The latest comment on this ticket mentions "...given the feedback we received from the 508 team I don't think we want these accordion items to be read back as headers."  

What is the recommendation from 508 team and how will this affect or address how 508 will assess this component? In other words, will this component pass according to 508 assessment report?

**Comment by humancompanion-usds** [comment]
@caw310 - File a request with CAIA for support here please. Thanks.

**Comment by caw310** [comment]
CAIA issue for a11y folks to investigate. https://github.com/department-of-veterans-affairs/va.gov-team/issues/61996

**Comment by caw310** [comment]
@imelda-guzman , is there a link you could provide for a11y folks?

**Comment by imelda-guzman** [comment]
Hi @caw310 sorry for the late response.

Our sites are behind a firewall and can only be accessed if you are provisioned to one or more VistA sites.  It can be a lengthy process to get access if you don't already have it.  Let me see if I can find an example on the va.gov site and get back to you.

**Comment by imelda-guzman** [comment]
@caw310 my apologies for not getting back to you on this ticket.  Our project has gone through another review since this ticket was first opened and the `va-accordion` looks to have passed 508.  The versions and markup of the code were updated to include the use of the prop `level` for the `va-accordion-item`.  We were not using this prop before.

Using the prop `level` gave us the ability to define the heading of the button therefore keeping the heading hierarchy of the content in correct order.  We suspect this change is what helped for passing 508.

I think it is safe to say this bug ticket no longer applies and ticket can be marked as resolved.
